src/reanimated2/initializers.ts

```diff
@@ -0,0 +1,150 @@
+import NativeReanimatedModule from './NativeReanimated';
+import { runOnUI, runOnJS } from './threads';
+
+function callGuard<T extends Array<any>, U>(
+  fn: (...args: T) => U,
+  ...args: T
+): void {
+  'worklet';
+  try {
+    fn(...args);
+  } catch (e) {
+    if (global.ErrorUtils) {
+      global.ErrorUtils.reportFatalError(e);
+    } else {
+      throw e;
+    }
+  }
+}
+
+function valueUnpacker(objectToUnpack: any, category?: string): any {
+  'worklet';
+  let workletsCache = global.__workletsCache;
+  let handleCache = global.__handleCache;
+  if (workletsCache === undefined) {
+    // init
+    workletsCache = global.__workletsCache = new Map();
+    handleCache = global.__handleCache = new WeakMap();
+  }
+  if (objectToUnpack.__workletHash) {
+    let workletFun = workletsCache.get(objectToUnpack.__workletHash);
+    if (workletFun === undefined) {
+      const evalFn =
```

Comment:
We could also memoize `evalFn` (i.e. initialize it analogously as `workletsCache`).

Reply:
don't see a clear benefit for memoizing this, this just a single if/else check that we need to perform so memoizing seem like a premature optimization